pkg/agent/utils.go

```diff
@@ -358,18 +358,44 @@ func GetOrderedKubeletConfigFlagString(k map[string]string, cs *datamodel.Contai
 	kubeletConfigFileEnabled := IsKubeletConfigFileEnabled(cs, profile, kubeletConfigFileToggleEnabled)
 	keys := []string{}
 	for key := range k {
-		if !kubeletConfigFileEnabled || !TranslatedKubeletConfigFlags[key] {
-			keys = append(keys, key)
+		// TODO(qinhao): need to consider the case when enable-kubelet-config-file is enabled
+		if kubeletConfigFileEnabled && TranslatedKubeletConfigFlags[key] {
```

Comment:
Custom configuration includes not only the configuration of kubelet but also kube URL, as well as other components' images and configurations for internal use. These will be a conflict with [custom node configuration](https://docs.microsoft.com/en-us/azure/aks/custom-node-configuration), but the two features are not expected to be used for the same cluster.
TranslatedKubeletConfigFlags is introduced for dyanmic kubelet by https://github.com/Azure/AgentBaker/pull/139/files, and has been disabled by @anujmaheshwari1 in AKS